### PR TITLE
do not mock `RequestStack` in tests

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 7.1
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | 
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Tests/EventListener/LocaleListenerTest.php [hunk 1]
```diff
@@ -26,11 +25,11 @@
 
 class LocaleListenerTest extends TestCase
 {
-    private MockObject&RequestStack $requestStack;
+    private RequestStack $requestStack;
 
     protected function setUp(): void
     {
-        $this->requestStack = $this->createMock(RequestStack::class);
+        $this->requestStack = new RequestStack();
```
Comment (OskarStark) [inline src/Symfony/Component/HttpKernel/Tests/EventListener/LocaleListenerTest.php]: Why this setup, I would use new RequestStack() directly for discoverability, no?

**Comment by OskarStark** [comment]
As this hardens the codebase, I am for target branch 5.4

**Review by nicolas-grekas — APPROVED** [review]
Thanks!
This is not a bugfix so 7.1 is correct to me. Also the code is different in 5.4 so moving the patch to it would double the effort.

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**